proposal/first-class-mixins.md

```diff
@@ -0,0 +1,199 @@
+# First Class Mixins: Draft 1
+
+*([Issue](https://github.com/sass/sass/issues/626))*
+
+This proposal promotes mixins to first-class values and adds members to the
+`sass:meta` module for manipulating them.
+
+## Table of Contents
+
+* [Background](#background)
+* [Summary](#summary)
+* [Types](#types)
+  * [Operations](#operations)
+    * [Equality](#equality)
+  * [Serialization](#serialization)
+* [Functions](#functions)
+  * [`meta.type-of()`](#metatype-of)
+  * [`meta.get-mixin()`](#metaget-mixin)
+  * [`meta.module-mixins()`](#metamodule-mixins)
+  * [`meta.accepts-content()`](#metaaccepts-content)
+* [Mixins](#mixins)
+  * [`meta.apply()`](#metaapply)
+## Background
+
+> This section is non-normative.
+
+Composability is a very powerful feature typically seen in functional languages
+and scripting languages. Sass partially supports this, but only for one of its
+two kinds of callables.
+
+## Summary
+
+> This section is non-normative.
+
+This proposal promotes mixins to first-class values, giving Sass composability
+for both of its kinds of callables.
+
+Additionally, it provides ways to interact with mixin values similarly to how
+one would with function values:
+
+* `get-function()` → `get-mixin()`
+
+* `module-functions()` → `module-mixins()`
+
+* `call()` → `apply()`
+
+## Types
+
+This proposal introduces a new value type known as a "mixin", with the
+following structure:
+
+```ts
+interface Mixin {
+  /**
+   * The name of this mixin
+   */
+  name: string;
+
+  /**
+   * Whether or not the body of this mixin contains the `@content` rule
+   */
+  hasContent: boolean;
+}
+```
+
+> The internal representation of a mixin may also include function arguments, a body, a span, or other fields, but these properties are not meaningfully exposed to users.
+>
+> When a mixin value is created, it captures the current execution environment and a pointer to the mixin object being referenced. This is also not directly exposed to users.
+
+### Operations
+
+The only operation permitted for mixin objects is checking for equality. 
+
+#### Equality
+
+Two mixins are considered equal if they refer to the exact same mixin as declared in the source code.
```

Comment:
What about mixins that aren't declared in source code, such as `meta.load-css()`?